[backport] :book: improve developer doc (#4264)
kubernetes-sigs/cluster-api: docs/book/src/clusterctl/developers.md

to make the controller images available for the kubelet in the management cluster.

When the kind cluster is ready and all the required images are in place, run

> the clusterctl init command generated by the create-local-repository.py

Reviewer: ```suggestion
the `clusterctl init` command generated by the `create-local-repository.py`
```

script.

Optionally, you may want to check if the components are running properly. The

> exactly components are depend on which providers you have initialized, below

Reviewer: ```suggestion
exact components depend on which providers you have initialized. Below
```

> is an example output with docker provider being installed.

Reviewer: ```suggestion
is an example with an installed Docker provider.
```

Author: done
